### PR TITLE
chore: fix delivery option text for publish page

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

Updates publish page to handle API mode.

Fixes https://github.com/cds-snc/platform-forms-client/issues/4726

### CODE REVIEW

File: i18n/translations/en/form-builder.json [hunk 1]
```diff
@@ -553,6 +553,7 @@
     "text2": "",
     "vaultOption": "Responses to be downloaded",
     "emailOption": "Responses delivered by email — ",
+    "apiOption": "Responses delivered by API",
```
Comment (timarney) [inline i18n/translations/en/form-builder.json:556]: @anikbrazeau here is the string for the publish page + FR

File: i18n/translations/fr/form-builder.json [hunk 2]
```diff
@@ -553,6 +553,7 @@
     "text2": "",
     "vaultOption": "Réponses à télécharger",
     "emailOption": "Réponses envoyées par courriel — ",
+    "apiOption": "Responses delivered by API [FR]",
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:556]:
```suggestion
    "apiOption": "Réponses récupérées avec l'API",
```

File: i18n/translations/en/form-builder.json (hunk 1 above)
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:556]:
```suggestion
    "apiOption": "Responses retrieved by API",
```

**Review by thiessenp-cds — APPROVED** [review]
LGTM :)

**Comment by anikbrazeau** [comment]
**In testing found:** It works going from Download to API and back. But shows Email when switching from Email to API.

**Comment by timarney** [comment]
> In testing found: It works going from Download to API and back. But shows Email on publishing checklist when switching from Email to API.

Will merge this PR and create a follow-up.

**Pull request merged**